wts.go

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) 2014-2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the license is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gowin32
+
+import (
+	"github.com/winlabs/gowin32/wrappers"
+
+	"fmt"
+	"net"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+// WTSConnectState enum type - Go version of WTS_CONNECTSTATE_CLASS
+type WTSConnectState uint32
+
+const (
+	WTSConnectStateActive       WTSConnectState = wrappers.WTSActive
+	WTSConnectStateConnected    WTSConnectState = wrappers.WTSConnected
+	WTSConnectStateConnectQuery WTSConnectState = wrappers.WTSConnectQuery
+	WTSConnectStateShadow       WTSConnectState = wrappers.WTSShadow
+	WTSConnectStateDisconnected WTSConnectState = wrappers.WTSDisconnected
+	WTSConnectStateIdle         WTSConnectState = wrappers.WTSIdle
+	WTSConnectStateListen       WTSConnectState = wrappers.WTSListen
+	WTSConnectStateReset        WTSConnectState = wrappers.WTSReset
+	WTSConnectStateDown         WTSConnectState = wrappers.WTSDown
+	WTSConnectStateInit         WTSConnectState = wrappers.WTSInit
+)
+
+// WTSClientProtocolType enum type go version of WTSClientProtocolType
+type WTSClientProtocolType uint32
+
+const (
+	WTSClientProtocolConsoleSession WTSClientProtocolType = 0
+	WTSClientProtocolInternal       WTSClientProtocolType = 1
+	WTSClientProtocolRDP            WTSClientProtocolType = 2
+)
+
+// WTSClientInfo - go version of WTSCLIENT structure
+type WTSClientInfo struct {
+	ClientName          string
+	Domain              string
+	UserName            string
+	WorkDirectory       string
+	InitialProgram      string
+	EncryptionLevel     byte
+	ClientAddressFamily AddressFamily
+	ClientAddress       [wrappers.CLIENTADDRESS_LENGTH + 1]uint16
+	HRes                uint16
+	VRes                uint16
+	ColorDepth          uint16
+	ClientDirectory     string
+	ClientBuildNumber   uint32
+	ClientHardwareId    uint32
+	ClientProductId     uint16
+	OutBufCountHost     uint16
+	OutBufCountClient   uint16
+	OutBufLength        uint16
```

Comment:
We generally expose fields on this side as `int` or `uint` rather than with an explicit size since that's what Go developers are most likely to be using in their code.

Reply:
Right.  I changed uint16 and uint32 to uint. I left only the sessionId as uint32.  What do you think ?